ipatests/test_xmlrpc/test_schema_plugin.py

```diff
@@ -0,0 +1,143 @@
+# Authors:
+#   Felipe Barreto <fbarreto@redhat.com>
+#
+# Copyright (C) 2012, 2017  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test the `ipaserver/plugins/schema.py` module.
+"""
+
+import pytest
+
+from ipalib import errors
+from ipatests.test_xmlrpc.tracker.base import Tracker
+from ipatests.test_xmlrpc.xmlrpc_test import XMLRPC_test
+
+
+@pytest.mark.tier1
+class TestParamFindAndShowCommand(XMLRPC_test):
+    """Test functionality of the ipa param-{find,show} command"""
+    tracker = Tracker()
+
+    def test_param_find(self):
+        """Test param-find command"""
+
+        # right command without criteria
+        criteria = u'postalcode'
+        result = self.tracker.run_command('param_find', u'user-add')
+        assert len(result['result']) != 0, result
+
+        # right command, right criteria
+        criteria = u'postalcode'
+        result = self.tracker.run_command('param_find', u'user-add', criteria)
+        assert len(result['result']) != 0, result
+        for item in result['result']:
+            assert (criteria in item['name'].lower() or
+                    criteria in item['doc'].lower()), item
+
+        # wrong command, right criteria
+        result = self.tracker.run_command('param_find', u'fake', criteria)
+        assert len(result['result']) == 0, result
+
+        # right command, wrong criteria
+        result = self.tracker.run_command('param_find', u'user-add', u'fake')
+        assert len(result['result']) == 0, result
+
+        # wrong command, wrong criteria
```

Comment:
Same here. Maybe this is legitimate but I'd think that once the command is determined to be bad it doesn't matter what else was passed, right? I mean, how can one determine the "right" criteria for a command without having the command defined? In this case, postalcode makes sense for users but not for groups, so isn't correct in all cases anyway.